pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -281,6 +283,34 @@ func (p *PriorityQueue) Run() {
 func (p *PriorityQueue) Add(pod *v1.Pod) error {
 	p.lock.Lock()
 	defer p.lock.Unlock()
+
+	if err := p.add(pod); err != nil {
+		return err
+	}
+
+	p.cond.Broadcast()
+	return nil
+}
+
+func (p *PriorityQueue) MoveUpPods(pods []*v1.Pod) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+
+	moved := false
+	for _, pod := range pods {
+		if err := p.add(pod); err == nil {
```

Comment:
What we want is to add the pod iif it was in backoff/unschedulable, using `add` means any pod could be added. What if the pod was deleted by the time we add it here?

Reply:
Good point. We'd need to add a safety check here as users may pass in illegal pods.